test/type.coffee

```diff
@@ -8,19 +8,33 @@ Amen.describe "Type functions", (context) ->
     isString, isFunction, isObject, isArray, isRegExp, isDate, isDefined,
     isGenerator, isPromise} = require "../src/type"
 
-  context.test "type"
-
-  context.test "isType"
-
-  context.test "instanceOf"
+  context.test "type", ->
+    assert type 7
```

Comment:
The `type` function just returns the `constructor` property of an object. This is frowned upon in favor of `typeof` and/or some additional processing (see [Crockford's implementation][0]). The reason is that, since `constructor` is an ordinary property, you can't trust it.

[0]:http://javascript.crockford.com/remedial.html

Fairmont's stance is that if people are going to change the constructor function, so be it, but we're not going to build around what amounts of an edge case. To wit:

```coffee
assert 7.constructor == Number
7.constructor = Array
assert 7.constructor == Number
```

but 

```coffee
class A
class B
a = new A
assert a.constructor == A
a.constructor = B
assert a.constructor == B
```

For built-in types, JavaScript makes `constructor` immutable. (I'm not 100% sure this true for all the built-in types, actually.) For everything else, _caveat emptor_. Sometimes you won't be able to use `type`.

Fairmont encourages currying `type` to create specialized type-checking functions, which is sort of a way of saying, “I am pretty sure this works for this type.”